generators/csharp/sdk/src/endpoint/http/HttpEndpointGenerator.ts

```diff
@@ -312,4 +309,347 @@ export class HttpEndpointGenerator extends AbstractEndpointGenerator {
             });
         });
     }
+
+    public generateHttpPagerMethod({
+        serviceId,
+        endpoint,
+        rawClientReference,
+        rawClient
+    }: {
+        serviceId: ServiceId;
+        endpoint: HttpEndpoint;
+        rawClientReference: string;
+        rawClient: RawClient;
+    }): csharp.Method {
+        this.assertHasPagination(endpoint);
+        const endpointSignatureInfo = this.getEndpointSignatureInfo({ serviceId, endpoint });
+        const parameters = [...endpointSignatureInfo.baseParameters];
+        const optionsParamName = this.getRequestOptionsParamNameForEndpoint({ endpoint });
+        const requestOptionsParam = this.getRequestOptionsParameter({ endpoint });
+        const requestOptionsType = requestOptionsParam.type;
+        parameters.push(requestOptionsParam);
+        const itemType = this.getPaginationItemType(endpoint);
+        const return_ = this.getPagerReturnType(endpoint);
+        const snippet = this.getHttpPagerMethodSnippet({ endpoint });
```

Comment:
There's a lot of parameters required here - I wonder if it's be worth rolling out a `PaginatedEndpointGenerator` class (or something similar) that could define all these properties as member variables, so that the following implementation could be reduced / a little more concise and readable.

Not totally necessary to do that here, but let me know your thoughts.